### ISSUE TITLE
Reverse Engineering - SqlException: The parameterized query expects the parameter '@returnValue', which was not supplied.

### ISSUE THREAD

**Issue by phil000** [ErikEJ/EFCorePowerTools#2161]
### Provide steps to reproduce a bug

When reverse engineering code that executes a stored procedure, the generated code causes an exception when executed:

*Microsoft.Data.SqlClient.SqlException: 'The parameterized query '(@TenantId int,@returnValue int output)EXEC @returnValue = [dbo]' expects the parameter '@returnValue', which was not supplied.'*

This is a change to how the code was previously generated (and working). If I revert the generated code to the old model it still works.

The change in new generation model causes exception:

![image](https://github.com/ErikEJ/EFCorePowerTools/assets/9067236/39202098-7758-4928-8c47-b869be99eaf7)

### Provide technical details

- EF Core Power Tools version:
**2.6.140**

- Exact Visual Studio version: (e.g. Visual Studio 2022 17.6.5)
**17.9.0**

- Database engine: (SQL Server, Azure SQL, Postgres, Oracle, Firebird, SQLite, MySQL)
**SQL Server**

- EF Core version in use: (e.g. EF Core 7)
**EF Core 8**

- Is Handlebars templates used: no/yes
**Yes**

- Is T4 templates used: no/yes
**No**

- Is a SQL Server .dacpac used: no/yes
**Yes**

**Comment by ErikEJ** [comment]
Since you are not providing any repro steps I am going to close this. Is this another case of now generating from a dacpac vs generating from a live database?

**Comment by phil000** [comment]
**Step 1:** Have a stored procedure in your project
**Step 2:** Use EF Core Power Tools -> Reverse Engineer menu item, using .dacpac Data Connection

**Current Behavior:**
Produces generated code that does not correctly execute stored procedure.

Microsoft.Data.SqlClient.SqlException: 'The parameterized query 'x' expects the parameter '@returnvalue', which was not supplied.'



The issue does *not* happen if I choose to 'reverse engineer' directly off the database. Having said that, we always try to generate off the model as that is what we have in source control and deploy.

**Comment by ErikEJ** [comment]
Still no sample stored procedure SQL script 😢

**Comment by phil000** [comment]
The issue is not stored procedure specific.

This stored procedure generates the same code, that produces the same issue.

```sql
CREATE PROCEDURE [dbo].[spGetAnything]
AS
BEGIN
	SET NOCOUNT ON;

	SELECT
		1 AS TrackableAssetStateVersion
END
GO
```

*Microsoft.Data.SqlClient.SqlException: 'The parameterized query '(@returnValue int output)EXEC @returnValue = [dbo].[spGetAnythin' expects the parameter '@returnValue', which was not supplied.'*

The generated code:
```C#
        public virtual async Task<List<spGetAnythingResult>> spGetAnythingAsync(OutputParameter<int> returnValue = null, CancellationToken cancellationToken = default)
        {
            var parameterreturnValue = new SqlParameter
            {
                ParameterName = "returnValue",
                Direction = System.Data.ParameterDirection.InputOutput,
                Value = returnValue?._value,
                SqlDbType = System.Data.SqlDbType.Int,
            };

            var sqlParameters = new []
            {
                parameterreturnValue,
            };
            var _ = await _context.SqlQueryAsync<spGetAnythingResult>("EXEC @returnValue = [dbo].[spGetAnything]", sqlParameters, cancellationToken);

            returnValue?.SetValue(parameterreturnValue.Value);

            return _;
        }
```

**Comment by ErikEJ** [comment]
Thanks for the sample - and did this suddenly happen in the latest release, or did it happen because you have switched to using the .dacpac instead of the live database?

**Comment by ErikEJ** [comment]
Ah - looks like you are the only user I have that generates and uses .dacpac stored procedures - it is a bug!

**Comment by phil000** [comment]
That seems pretty astonishing out of all the users. We use the .dacpac for generation as it is what is actually packaged and deployed from devops. So it seems better to rely on that as the 'master' of our database.

But as long as the issue is recorded.

**Comment by ErikEJ** [comment]
It could be a regression in the latest release, I will have a look

**Comment by ErikEJ** [comment]
I implemented a fix for this in the [latest daily build](https://www.vsixgallery.com/extension/f4c4712c-ceae-4803-8e52-0e2049d5de9f), would be grateful if you could try it out.

**Comment by phil000** [comment]
This is resolved in 2.6.143
